appservice/src/tree/DeploymentsTreeItem.ts

```diff
@@ -27,10 +27,11 @@ export class DeploymentsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     private readonly _connectToGitHubCommandId: string;
 
-    public constructor(parent: AzureParentTreeItem<ISiteTreeRoot>, siteConfig: SiteConfig, connectToGitHubCommandId: string) {
+    public constructor(parent: AzureParentTreeItem<ISiteTreeRoot>, siteConfig: SiteConfig, sourceControl: SiteSourceControl, connectToGitHubCommandId: string) {
         super(parent);
         this.contextValue = siteConfig.scmType === ScmType.None ? DeploymentsTreeItem.contextValueUnconnected : DeploymentsTreeItem.contextValueConnected;
```

Comment:
What do you think about adding things like 'repoUrl' and 'scmType' (or even `siteConfig` and `sourceControl`) to `ISiteRoot` or `SiteClient`? It seems like multiple different tree items are using those types of properties and I feel like it's inefficient to get them so many times. We'd probably have to add a 'refresh' function to ISiteRoot, though

Reply:
Yeah, I agree.  We tend to grab it a lot for deployment-related things.  I think I prefer having it as a property of `SiteClient` and having refresh just create a new, updated `client`.  Do you have a preference?

Reply:
I don't want to add anything to SiteClient that involves getting the plan due to https://github.com/Microsoft/vscode-azureappservice/issues/769, but site config and source control should be fine

Reply:
So I think it only makes sense to add `SiteConfig` to `SiteClient`.  `SourceControl` is only used three times and it's mostly related to Git deployment.  Furthermore, `SiteConfig` is included in the `Site` object, but `SourceControl` requires a separate API call.

Reply:
Never mind about `SiteConfig` being included in the `Site` object.  It's an optional parameter and isn't returned with whatever API call we used to get the `Site`.